types/tx_msg.go

```diff
@@ -79,6 +79,14 @@ type (
 		GetTimeoutHeight() uint64
 	}
 
+	// TxWithUnordered extends the Tx interface by allowing a transaction to set
+	// the unordered field, which implicitly relies on TxWithTimeoutHeight.
+	TxWithUnordered interface {
+		TxWithTimeoutHeight
```

Comment:
Is this true in the final result of unordered tx?  Do you need to implement a TimeoutHeight even though we use a timestamp?

This might just be because it is the initial commit - just curious